### PR TITLE
Add `tagpad` and `age_at_event` functions.

### PR DESCRIPTION
**Pull request by dmbates**
This PR adds a tagpad function as described in https://github.com/JuliaMixedModels/EmbraceUncertainty/issues/63. It also adds `age_at_event` and updates `Project.toml` to revise the `compat` entries and add the `Dates` package.

I have updated the :ELP_ldt_subj and :ELP_ldt_trial Arrow files on the osf.io site to use tagged strings for `subj`. I realized after I did so that we don't have a mechanism for updating the cached copies in the scratchspace directory, other than manually deleting the file then using `dataset` to force another download.

Perhaps we should add an optional argument `refresh::Bool = false` to the `dataset` function. When set to `true` it would force a download of the Arrow file from the osf.io site to the scratchspace directory.

### CODE REVIEW

**Comment by dmbates** [comment]
What do you think of adding a `refresh::Bool=true` argument to `dataset`?

File: src/EmbraceUncertainty.jl [hunk 1]
```diff
@@ -123,7 +130,45 @@ function readme()
...
+same order as the original numeric values.
+
+The single-argument version, e.g. `tagpad(:I)`, returns a partially-applied function that can be used
+in a `transform` or `select` call.
+
+```@example
+show(tagpad(repeat(1:10, inner=2)))
+```
+"""
+function tagpad(v::AbstractVector{<:Integer}, tag::String="S"; pool::Bool=true)
```
Comment (palday) [inline src/EmbraceUncertainty.jl:149]:
```suggestion
function tagpad(v::AbstractVector{<:Integer}, tag::AbstractString="S"; pool::Bool=true)
```

File: src/EmbraceUncertainty.jl [hunk 2]
```diff
@@ -123,7 +130,45 @@ function readme()
...
+    tagged = string.(tag, lpad.(v, maximum(ndigits, v), '0'))
```
Comment (palday) [inline src/EmbraceUncertainty.jl:150]: if you really wanted to, you could add one more method here

```julia
tagpad(v::Integer, tag::AbstractString, padlen::Integer) = string(tag, lpad(v, padlen, '0'))
```

Comment (dmbates) [reply src/EmbraceUncertainty.jl:150]: Good suggestion.  Thanks.

**Review by palday — APPROVED** [review]

**Comment by dmbates** [comment]
Added a named argument `reload` to the `dataset` function.  I am still debating if it is a good idea to allow updates to the data sets on the osf.io site.  On balance, I think it is good but it does open the possibility of a user getting a different data table than what we discuss in the book.  I plan to add a date stamp and, perhaps, a version number to the metadata for each of the data sets on the osf.io site and ask users to check the date stamp if they are having difficulties.  Does that seem reasonable?

**Comment by palday** [comment]
> Added a named argument `reload` to the `dataset` function. I am still debating if it is a good idea to allow updates to the data sets on the osf.io site. On balance, I think it is good but it does open the possibility of a user getting a different data table than what we discuss in the book. I plan to add a date stamp and, perhaps, a version number to the metadata for each of the data sets on the osf.io site and ask users to check the date stamp if they are having difficulties. Does that seem reasonable?

That seems reasonable

**Comment by palday** [comment]
closes #63

**Pull request merged**